### ISSUE TITLE
Sign

### ISSUE THREAD

**Issue by simranagarwal54** [cryptotronix/cryptoauth-arduino#13]
Hi,
I am using this library to work on Sparkfun Cryptoshield and Arduino Uno. 
Currently, I am working the basic encryption and decryption using the AtECC108a chip on the same.
The steps I used are-
1. Wakeup
2. GenKey command in Private mode.
3. Random Seed Update.
4. Create Message Digest Using sha256 function.
5. Nonce command to pass the Message Digest
6. Use sign command with the same Key ID as that of the private key in GenKey command
7. Verify command where I pass the output parameter of the GenKey command along with the output parameter of the sign command. The message digest is passed to the chip using the nonce command.
8. Sleep 
Problems-
1. The verification command fails in this sequence. All other commands are successfully executed.
2. I am not sure if the Sign command is using the private key generated by the GenKey command. I would want it to do so and how can I ensure the same.
3. What is the work of the Random seed update?
4. My objective is to generate the Private-Public Key pair and use the private key for signing and it's public key for verification purpose, But I can't figure a way to generate a private-public key pair or use it in the correct way.

Any information on the usage of this chip would be extremely appreciated.
Thank you in advance.

**Comment by jbdatko** [comment]
Yeah, so that all sounds correct. My arduino driver is really not very good though. Microchip has a driver for the 108/508 that has this sign and verify command. Microchip changed the website up so I couldn't find on a quick search. But you _really_ want to use that as it's much better.

It doesn't work with arduino though but does work with Atmel Studio. So there would be some porting to do if you need it work in the Arduino environment.

**Comment by simranagarwal54** [comment]
Thank you for your time and Information.
I have implemented the verification using this library itself. I had to make certain changes in the code, but this library was of great help.

However, could you also help me out with any guidance/knowledge about generating digital certificates using the ATECC108 chip and Arduino UNO? 
I mean Is it possible to generate a certificate for the verification process using Arduino Uno that can be accessible by the above mentioned chip?

**Comment by jbdatko** [comment]
So, you _could_ but you really don't want to do it that way. First of all, you'd need the NDA document. Secondly, even if you had it the process is very convoluted. 

The problem is that certificates are variable length and even if you have a basic one, you are still looking at about 800 or so bytes. The 328p is so very limited too. So keep the entire cert either in an external EEPROM with an attached signature or maybe it'll fit in the internal EEPROM of the 328p? Or, just keep a signed hash of the certificate around and get the certificate on demand.

It's really hard to say without knowing your design constraints.

**Comment by simranagarwal54** [comment]
I actually have no Idea how to generate a certificate for the signing/verification purpose. 
As per now I have no design constraints, Only that the certificate created will be used for the verification of the signature using the verify_hash function in the cryptoauth library. only that instead of sending the public key directly, the certificate will be sent.

But, how should I create a digital certificate that can store the public key of the slot used for signing in ATECC108 is the real problem that I am facing?

I really appreciate you sparing some time for my issue. Thank you.
